saagieapi/projects/saagie_api.py

```diff
@@ -1134,6 +1138,183 @@ def get_project_app(self, app_id):
+
+        Returns
+        -------
+        dict
+            Dict of app information
+        """
+
+        params = {
+            "projectId": project_id,
+            "name": app_name,
+            "description": description,
+            "releaseNote": release_note,
+            "storageSizeInMB": storage_size_in_mb,
+            "image": image,
+            "exposedPorts": exposed_ports,
+            "storagePaths": storage_paths,
+        }
+        check_fomat_exposed_port = check_exposed_ports(exposed_ports, self.list_exposed_ports_field)
+        if not check_fomat_exposed_port:
+            raise ValueError(
+                f"The parameter 'exposed_ports' should be a list of dict. Each dict should contains the key 'port'."
+                "All accept key of each dict is: '{list_exposed_port_field}'")
+
+        all_technologies_in_catalog = [
+            catalog['technologies'] for catalog in self.get_repositories_info()['repositories']
+            if catalog['name'] == technology_catalog
+        ]
+        if not all_technologies_in_catalog:
+            raise RuntimeError(
+                f"Catalog {technology_catalog} does not exist or does not contain technologies")
+        technology_in_catalog = [tech['id'] for tech in
+                                 all_technologies_in_catalog[0]
+                                 if
+                                 tech["label"].lower() == technology.lower() and tech["__typename"] == "AppTechnology"]
+
+        if not technology_in_catalog:
+            raise RuntimeError(
+                f"Technology {technology} does not exist in the catalog {technology_catalog}")
+        technology_id = technology_in_catalog[0]
+        params["technologyId"] = technology_id
+
+        if docker_credentials_id:
+            params["dockerCredentialsId"] = docker_credentials_id
+
+        if emails:
+            wrong_status_list = []
+
+            for item in status_list:
+                if item not in self.valid_status_list:
+                    wrong_status_list.append(item)
+            if wrong_status_list:
+                raise RuntimeError(f"The following status are not valid: {wrong_status_list}. "
+                                   f"Please make sure that each item of the parameter status_list should be "
+                                   f"one of the following values: 'REQUESTED', 'QUEUED', 'RUNNING', "
+                                   f"'FAILED', 'KILLED', 'KILLING', 'SUCCEEDED', 'UNKNOWN', 'AWAITING', 'SKIPPED'")
+
+            else:
+                params["alerting"] = {
+                    "emails": emails,
+                    "statusList": status_list
+                }
+        query = gql(gql_create_app)
+        return self.client.execute(query, variable_values=params)
+
+    def edit_app(self, app_id, app_name=None, description=None, emails=None, status_list=["FAILED"]):
+        """Edit an app
```

Comment:
Maybe describe here what is possible to edit in an App

Reply:
corrected in the commit [refactoring of app functions](https://github.com/saagie/api-saagie/pull/81/commits/9fb6fc8e9ca05d9ead5a048dca172f15a719ade2)